### PR TITLE
0.10 dev

### PR DESCRIPTION
**Pull request by zhangyi999**
add coin ID

### CODE REVIEW

**Review by dindinw — CHANGES_REQUESTED** [review]
Instead of master, Please PR to 0.10-dev branch (https://github.com/Qitmeer/qitmeer.js/tree/0.10)

File: src/transaction.js [hunk 1]
```diff
@@ -386,8 +391,8 @@ Transaction.prototype.hashForSignature = function (inIndex, prevOutScript, hashT
     //    d) 4 bytes sequence
     // 4) number of outputs varint
     // 5) per output:
+    //    0) 2 bytes coinId
```
Comment (dindinw) [inline src/transaction.js:394]: //    a) 2 bytes coinId
    //    b) 8 bytes amount

File: src/transaction.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ Transaction.prototype.byteLength = function (stype) {
     (onlyWitnesses ? 0 : varuint.encodingLength(this.vin.length)) +
     (onlyWitnesses ? 0 : varuint.encodingLength(this.vout.length)) +
     (onlyWitnesses ? 0 : this.vin.reduce(function (sum, input) { return sum + 32 + 4 + 4 }, 0)) + // txid + vout + seq
-    (onlyWitnesses ? 0 : this.vout.reduce(function (sum, output) { return sum + 8 + varSliceSize(output.script) }, 0)) + // amount + script
+    (onlyWitnesses ? 0 : this.vout.reduce(function (sum, output) { return 4 + sum + 8 + varSliceSize(output.script) }, 0)) + // coinId + amount + script
```
Comment (dindinw) [inline src/transaction.js:156]: 2? 4?

**Comment by zhangyi999** [comment]
Need to resubmit

**Comment by zhangyi999** [comment]
Need to resubmit

**Pull request closed without merging**